static/app/components/profiling/FlamegraphZoomView.tsx

```diff
@@ -0,0 +1,540 @@
+      !scheduler ||
+      !textRenderer ||
+      !gridRenderer
+    ) {
+      return undefined;
+    }
+
+    function drawSelectedFrameBorder() {
+      // We are doing this because of typescript, in reality we are creating a scope
+      // where flamegraphRenderer is not null (see check on L86)
+      if (
+        flamegraphRenderer === null ||
+        textRenderer === null ||
+        selectedFrameRenderer === null
+      ) {
+        return;
+      }
+
+      // If no node is selected or hovered, then dont draw anything
+      if (!selectedNode && !hoveredNode) {
+        return;
+      }
+
+      textRenderer.context.clearRect(
+        0,
+        0,
+        flamegraphRenderer.physicalSpace.width,
+        flamegraphRenderer.physicalSpace.height
+      );
+
+      if (selectedNode) {
+        selectedFrameRenderer.draw(
+          new Rect(
+            selectedNode.start,
+            flamegraph.inverted
+              ? flamegraphRenderer.configSpace.height - selectedNode.depth
+              : selectedNode.depth,
+            selectedNode.end - selectedNode.start,
+            1
+          ),
+          {
+            BORDER_COLOR: flamegraphRenderer.theme.COLORS.SELECTED_FRAME_BORDER_COLOR,
+            BORDER_WIDTH: flamegraphRenderer.theme.SIZES.FRAME_BORDER_WIDTH,
+          },
+          selectedFrameRenderer.context,
+          flamegraphRenderer.configToPhysicalSpace
+        );
+      }
+      if (hoveredNode && selectedNode !== hoveredNode) {
+        selectedFrameRenderer.draw(
+          new Rect(
+            hoveredNode.start,
+            flamegraph.inverted
+              ? flamegraphRenderer.configSpace.height - hoveredNode.depth
+              : hoveredNode.depth,
+            hoveredNode.end - hoveredNode.start,
+            1
+          ),
+          {
+            BORDER_COLOR: flamegraphRenderer.theme.COLORS.HOVERED_FRAME_BORDER_COLOR,
+            BORDER_WIDTH: flamegraphRenderer.theme.SIZES.HOVERED_FRAME_BORDER_WIDTH,
+          },
+          selectedFrameRenderer.context,
+          flamegraphRenderer.configToPhysicalSpace
+        );
+      }
+
+      textRenderer.draw(
+        flamegraphRenderer.configView,
+        flamegraphRenderer.configSpace,
+        flamegraphRenderer.configToPhysicalSpace
+      );
+    }
+
+    scheduler.registerBeforeFrameCallback(drawSelectedFrameBorder);
+    scheduler.draw();
+
+    return () => {
+      scheduler.unregisterBeforeFrameCallback(drawSelectedFrameBorder);
+    };
+  }, [
+    textRenderer,
+    selectedFrameRenderer,
+    selectedNode,
+    hoveredNode,
+    scheduler,
+    flamegraphRenderer,
+    flamegraph,
+  ]);
+
+  const onCanvasClick = React.useCallback(
+    (evt: React.MouseEvent<HTMLCanvasElement>) => {
+      evt.preventDefault();
+      evt.stopPropagation();
+
+      if (!flamegraphRenderer || !configSpaceCursor) {
+        return;
+      }
+
+      const newSelectedNode = flamegraphRenderer.getHoveredNode(configSpaceCursor);
```

Comment:
Is this equivalent to the `hoveredNode` state var? Can we reuse that instead of calling `getHoveredNode` again?